os/kernel/log_dump/log_dump.h

```diff
@@ -0,0 +1,178 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __LOG_DUMP_LOG_DUMP_H
+#define __LOG_DUMP_LOG_DUMP_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <queue.h>
+#include <mqueue.h>
+#include <stdbool.h>
+
+/****************************************************************************
+ * Pre-Processor Definitions
+ ****************************************************************************/
+
+#define LOG_DUMP_OK_HEAPCHK		1	/* To check heap available after success return */
+#define LOG_DUMP_MEM_FAIL	-1
```

Comment:
It is same value as LOG_DUMP_COMPRESS_FAIL.
Please change and move it to line 43 for readability.